### DISCUSSION TITLE
Bringing up ulab on a Pyboard v1.1

### DISCUSSION THREAD

**Discussion by grodstein** [micropython/micropython#11265] [STM32 / Pyboard] [answered]
I'm trying to bring up ulab on my Pyboard v1.1, with little success so far. I've tried:
- go to https://github.com/v923z/micropython-builder/releases to find a pre-built .dfu. However, while they do list pybv10.dfu, I don't see pybv11.dfu. I tried flashing my Pyboard with pybv10.dfu and, unsurprisingly, it didn't work.
- go to https://github.com/v923z/micropython-ulab#stm-based-boards and follow the directions there to cross-compile my own .dfu file for a Pyboard from RHEL. No luck there either; I just got lots of errors about the ARM cross-compiler not being able to find stdint.h. I did check, and the file exists.

So: am I missing some obvious way of getting a prebuilt .dfu? I would have thought that a Pyboard v1.1 is a very standard target, and am a bit surprised that it's not available prebuilt.
And: does anyone have any clue what's going wrong with my attempt to build the firmware myself? I can always try to reverse engineer the makefile, but hopefully many people have seen this error before. You can see a detailed log of the process, including the errors, at http://www.eecs.tufts.edu/~joelg/ulab_build_errors.txt

And sorry if, as perhaps expected, I'm doing something really stupid: while I have tons of experience at CPU design and post-silicon debug and even biology :-), I unfortunately can't count Linux sysadmin or embedded design as my strengths (but my sysadmin guy is quite good).
Thanks,
/Joel

**Comment by rkompass** [comment, marked as answer]
Perhaps you missed to let you show all assets on v923z's github site? Pyboard 1.1 is there: [https://github.com/v923z/micropython-builder/releases/download/latest/PYBV11.dfu](https://github.com/v923z/micropython-builder/releases/download/latest/PYBV11.dfu)

**Reply by grodstein** [reply]
Quite correct! Now that I clicked the "show all 13 assets" button, the Pyboard v1.1 firmware shows up, and flashes just fine :-). So, the immediate problem is solved.
For the longer term, it would still be nice to be able to compile up my own firmware version. Has anybody seen this particular firmware-build error (not being able to find standard-library files) before?
Thanks,
/Joel

**Reply by v923z** [reply]
The build process on github is the same as what you would have on a local PC, so you can just copy the scripts, and run them on the command line.
